packages/core/content-type-builder/admin/src/components/FormModal/forms/index.js

```diff
@@ -5,8 +5,84 @@ import { categoryForm, createCategorySchema } from '../category';
 import { contentTypeForm, createContentTypeSchema } from '../contentType';
 import { createComponentSchema, componentForm } from '../component';
 import { dynamiczoneForm } from '../dynamicZone';
+import { nameField } from '../attributes/nameField';
+import addItemsToFormSection from './utils/addItemsToFormSection';
+import getTrad from '../../../utils/getTrad';
+
+const getUsedAttributeNames = (attributes, schemaData) => {
+  return attributes
+    .filter(({ name }) => {
+      return name !== schemaData.initialData.name;
+    })
+    .map(({ name }) => name);
+};
 
 const forms = {
+  customField: {
+    schema({
+      schemaAttributes,
+      attributeType,
+      customFieldValidator,
+      reservedNames,
+      schemaData,
+      ctbFormsAPI,
+    }) {
+      const usedAttributeNames = getUsedAttributeNames(schemaAttributes, schemaData);
+
+      const attributeShape = attributeTypes[attributeType](
+        usedAttributeNames,
+        reservedNames.attributes
+      );
+
+      return ctbFormsAPI.makeCustomFieldValidator(
+        attributeShape,
+        customFieldValidator,
+        usedAttributeNames,
+        reservedNames.attributes,
+        schemaData
+      );
+    },
+    form: {
+      base({ customField }) {
+        // Default section with required name field
+        const sections = [{ sectionTitle: null, items: [nameField] }];
+
+        if (customField.options?.base) {
+          addItemsToFormSection(customField.options.base, sections);
+        }
+
+        return { sections };
+      },
+      advanced({ customField, data, step, extensions, ...rest }) {
+        // Default section with no fields
+        const sections = [{ sectionTitle: null, items: [] }];
+        const injectedInputs = extensions.getAdvancedForm(['attribute', customField.type], {
+          data,
+          type: customField.type,
+          step,
+          ...rest,
+        });
+
+        if (customField.options?.advanced) {
+          addItemsToFormSection(customField.options.advanced, sections);
+        }
+
+        if (injectedInputs) {
```

Comment:
Where is that logic handled for non-custom field attributes?

Reply:
I believe it is this: https://github.com/strapi/strapi/blob/aaf64cefa0365ccc900a89a978b502a8ccb77ba6/packages/core/content-type-builder/admin/src/components/FormModal/forms/index.js#L132-L148